framework/web/js/source/jquery.ba-bbq.min.js

```diff
@@ -1,23 +1 @@
es[val]:val),keys_last)for(;i<=keys_last;i++){if(key=""===keys[i]?cur.length:keys[i],prohibited_keys.includes(key))return;cur=cur[key]=i<keys_last?cur[key]||(keys[i+1]&&isNaN(keys[i+1])?{}:[]):val}else $.isArray(obj[key])?obj[key].push(val):void 0!==obj[key]?obj[key]=[obj[key],val]:obj[key]=val;else key&&(obj[key]=coerce?void 0:"")}),obj},jq_deparam.querystring=curry(jq_deparam_sub,0),jq_deparam.fragment=jq_deparam_fragment=curry(jq_deparam_sub,1),$.elemUrlAttr||($.elemUrlAttr=function(obj){return $.extend(elemUrlAttr_cache,obj)})({a:"href",base:"href",iframe:"src",img:"src",input:"src",form:"action",link:"href",script:"src"}),jq_elemUrlAttr=$.elemUrlAttr,$.fn.querystring=curry(jq_fn_sub,"querystring"),$.fn.fragment=curry(jq_fn_sub,"fragment"),jq_bbq.pushState=jq_bbq_pushState=function(params,merge_mode){is_string(params)&&/^#/.test(params)&&void 0===merge_mode&&(merge_mode=2);var has_args=void 0!==params,params=jq_param_fragment(location.href,has_args?params:{},has_args?merge_mode:2);location.href=params},jq_bbq.getState=jq_bbq_getState=function(key,coerce){return void 0===key||"boolean"==typeof key?jq_deparam_fragment(key):jq_deparam_fragment(coerce)[key]},jq_bbq.removeState=function(arr){var state={};void 0!==arr&&(state=jq_bbq_getState(),$.each($.isArray(arr)?arr:arguments,function(i,v){delete state[v]})),jq_bbq_pushState(state,2)},special.hashchange=$.extend(special.hashchange,{add:function(handleObj){var old_handler;function new_handler(e){var hash=e.fragment=jq_param_fragment();e.getState=function(key,coerce){return void 0===key||"boolean"==typeof key?jq_deparam(hash,key):jq_deparam(hash,coerce)[key]},old_handler.apply(this,arguments)}if($.isFunction(handleObj))return old_handler=handleObj,new_handler;old_handler=handleObj.handler,handleObj.handler=new_handler}})})(jQuery),(($,window)=>{var fake_onhashchange,timeout_id,last_hash,history_set,history_get,doc=document,special=$.event.special,doc=doc.documentMode,supports_onhashchange="onhashchange"in window&&(void 0===doc||7<doc);function get_fragment(url){return"#"+(url=url||location.href).replace(/^[^#]*#?(.*)$/,"$1")}function fn_retval(val){return val}function poll(){var hash=get_fragment(),history_hash=history_get(last_hash);hash!==last_hash?(history_set(last_hash=hash,history_hash),$(window).trigger("hashchange")):history_hash!==last_hash&&(location.href=location.href.replace(/#.*/,"")+history_hash),timeout_id=setTimeout(poll,$.fn.hashchange.delay)}$.fn.hashchange=function(fn){return fn?this.bind("hashchange",fn):this.trigger("hashchange")},$.fn.hashchange.delay=50,special.hashchange=$.extend(special.hashchange,{setup:function(){if(supports_onhashchange)return!1;$(fake_onhashchange.start)},teardown:function(){if(supports_onhashchange)return!1;$(fake_onhashchange.stop)}}),doc={},last_hash=get_fragment(),history_get=history_set=fn_retval,doc.start=function(){timeout_id||poll()},doc.stop=function(){timeout_id&&clearTimeout(timeout_id),timeout_id=void 0},fake_onhashchange=doc})(jQuery,this);
```

Comment:
The comment block from previous version is missing.

Reply:
The build comment removes the doc block. I added it back in manually.